python/cli/prelude_cli/views/build.py

```diff
@@ -114,12 +115,19 @@ def test_id():
 
     def upload(p: Path):
         with open(p, 'rb') as data:
-            with Spinner(description='Uploading to test'):
+            with Spinner(description='Uploading to test') as spinner:
                 data = controller.upload(
                     test_id=identifier, 
                     filename=p.name, 
                     data=data.read()
                 )
+                if 'compile_job_id' in data:
+                    spinner.update(spinner.task_ids[-1], description='Compiling')
+                    while (result := controller.get_compile_status(data['compile_job_id'])) and result['status'] == 'RUNNING':
+                        time.sleep(2)
+                    if result['status'] == 'FAILED':
+                        raise Exception('Failed to process attachment')
```

Comment:
since the test is still uploaded, maybe just set result to dict(error='failed to compile') or somethign and still merge data |= result ?